tests/unit/s2n_renegotiate_io_test.c

```diff
@@ -0,0 +1,483 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "tls/s2n_renegotiate.h"
+
+#include "tls/s2n_tls.h"
+#include "utils/s2n_safety.h"
+#include "utils/s2n_socket.h"
+
+enum S2N_TEST_APP_DATA_CASES {
+    S2N_TEST_APP_DATA_BEFORE_RENEG          = 1,
+    S2N_TEST_APP_DATA_BEFORE_SERVER_HELLO   = 2,
+    S2N_TEST_APP_DATA_AFTER_SERVER_HELLO    = 4,
```

Comment:
I read the test where these cases are used and I'm not sure why these values have to be powers of 2. What are you gaining here?

Reply:
OOOH you're testing combinations of cases. Doesn't this repeat several tests, like when "test_case" goes from 4-8?

Reply:
Discussed offline, we should keep it this way.

Reply:
I also added a comment-- see if that helps.